File: client/packages/domain/lib/src/viewing_appointment/entities/viewing_appointment_entity.dart
enum ViewingAppointmentStatus { pending, approved, rejected, cancelled }

abstract class ViewingAppointmentEntity {
  ViewingAppointmentEntity({
    required this.id,
    required this.tenantId,
    required this.roomId,
    required this.scheduledAt,
    required this.status,
    required this.createdAt,
    required this.updatedAt,
    this.note,
    this.roomTitle,
    this.roomAddress,
    this.roomMonthlyRent,
    this.hasRentalRequest = false,
  });

  final String id;
  final String tenantId;
  final String roomId;
  final String scheduledAt;
  final ViewingAppointmentStatus status;
  final String? note;
  final String createdAt;
  final String updatedAt;
  final String? roomTitle;
  final String? roomAddress;
  final String? roomMonthlyRent;
  final bool hasRentalRequest;
}

File: server/src/modules/viewing-appointments/domain/entities/viewing-appointment.entity.ts
export type ViewingAppointmentStatus =
  | 'pending'
  | 'approved'
  | 'rejected'
  | 'cancelled';

export class ViewingAppointmentEntity {
  constructor(
    public readonly id: string,
    public readonly tenantId: string,
    public readonly roomId: string,
    public readonly scheduledAt: Date,
    public readonly status: ViewingAppointmentStatus,
    public readonly note: string | null,
    public readonly createdAt: Date,
    public readonly updatedAt: Date,
    public readonly roomTitle?: string,
    public readonly roomAddress?: string,
    public readonly roomMonthlyRent?: string,
  ) {}
}
